ContractPlugin/CsharpContainer.cs

```diff
@@ -0,0 +1,81 @@
+using Google.Protobuf.Compiler;
+using Google.Protobuf.Reflection;
+
+// using Microsoft.CodeAnalysis.CSharp;
+// using static Microsoft.CodeAnalysis.SyntaxNode;
+
+namespace contract_plugin_csharp;
+
+// Generates the overall "container" for the generated C# contract
+public class CSharpContainer
```

Comment:
Let's make the name more meaningful. It's a generator for ContractContainer, we can call it `ContractContainerGenerator`. We can define an interface for `IGenerator<T>` and make T the input type of the descriptor.